### PR TITLE
Large-scale cleanup of scan and model

### PR DESCRIPTION
**Pull request by msoucy**

### CODE REVIEW

**Review by jeffaburke — APPROVED** [review]
Looks good to me

File: signinapp/event.py [hunk 1]
```diff
@@ -59,26 +70,39 @@ def selfevent():
...
     event = request.values.get("event_code")
     user_code = request.values.get("user_code")
 
-    ev: Event = db.session.scalar(select(Event).filter_by(code=event))
+    if not user_code:
+        return Response(
+            f"Error: Not a valid QR code: {user_code}", HTTPStatus.BAD_REQUEST
+        )
+    if not (user := User.from_code(user_code)):
+        return Response("Error: User does not exist", HTTPStatus.BAD_REQUEST)
```
Comment (bot190) [inline signinapp/event.py]: Should this be a 404 since the user wasn't found?

File: signinapp/event.py [hunk 2]
```diff
@@ -59,26 +70,39 @@ def selfevent():
...
+
+    if not user.approved:
+        return Response("Error: User is not approved", HTTPStatus.BAD_REQUEST)
```
Comment (bot190) [inline signinapp/event.py]: 403 here?

File: signinapp/templates/event.html.jinja2 [hunk 3]
```diff
@@ -32,17 +32,15 @@
         {%- if current_user.role.mentor -%}
           <div class="d-inline-flex justify-content-center m-auto container"
                id="mentorSelfSignIn">
+            <a class="btn btn-primary"
+              href="{{ url_for('event.selfscan', event_code=event_code) }}"
```
Comment (bot190) [inline signinapp/templates/event.html.jinja2]: Should this be `event.selfevent`?

Comment (msoucy) [reply signinapp/templates/event.html.jinja2]: 🤦‍♂️

File: signinapp/event.py (hunk 2 above)
Comment (msoucy) [reply signinapp/event.py]: I did 401 - from the `HTTPStatus` definition, 403 is described as "authorization would not help"

Comment (bot190) [reply signinapp/event.py]: Which is true. If they aren't approved, logging in again isn't going to fix anything here. 
`If authentication credentials were provided in the request, the server considers them insufficient to grant access`
https://httpwg.org/specs/rfc9110.html#status.403

**Review by bot190 — APPROVED** [review]

**Pull request merged**